packages/snap/tests/e2e/changePermission.spec.ts

```diff
@@ -0,0 +1,312 @@
+    const spyQuery = vi.spyOn(snapMock.rpcMocks, 'snap_dialog');
+
+    const defaultState = getDefaultSnapState(account);
+
+    defaultState[CURRENT_STATE_VERSION].config.dApp.permissions['random.io'] = {
+      ...getInitialPermissions(),
+      queryCredentials: true,
+    };
+
+    await snapMock.rpcMocks.snap_manageState({
+      operation: 'update',
+      newState: defaultState,
+    });
+
+    const res = (await onRpcRequest({
+      origin: 'https://random.io',
+      request: {
+        id: 'test-id',
+        jsonrpc: '2.0',
+        method: 'queryCredentials',
+        params: {},
+      },
+    })) as Result<unknown>;
+
+    if (isError(res)) {
+      throw new Error(res.error);
+    }
+
+    expect(res.data).toStrictEqual([]);
+
+    expect(spyQuery).toHaveBeenCalledTimes(0);
+
+    const stateOld = await snapMock.rpcMocks.snap_manageState({
+      operation: 'get',
+    });
+
+    expect(
+      stateOld[CURRENT_STATE_VERSION].config.dApp.permissions['random.io']
+    ).toStrictEqual({ ...getInitialPermissions(), queryCredentials: true });
+
+    const res2 = (await onRpcRequest({
+      origin: 'https://random.io',
+      request: {
+        id: 'test-id',
+        jsonrpc: '2.0',
+        method: 'changePermission',
+        params: {
+          origin: 'https://random.io',
+          method: 'queryCredentials',
+          value: false,
+        },
+      },
+    })) as Result<unknown>;
+
+    if (isError(res2)) {
+      throw new Error(res2.error);
+    }
+
+    expect(isSuccess(res2)).toBe(true);
+    expect(spyPermission).toHaveBeenCalledTimes(1);
+
+    const state = await snapMock.rpcMocks.snap_manageState({
+      operation: 'get',
+    });
+
+    expect(
+      state[CURRENT_STATE_VERSION].config.dApp.permissions['random.io']
+    ).toStrictEqual({ ...getInitialPermissions(), queryCredentials: false });
+
+    const spyQuery2 = vi.spyOn(snapMock.rpcMocks, 'snap_dialog');
+
+    const res3 = (await onRpcRequest({
+      origin: 'https://random.io',
+      request: {
+        id: 'test-id',
+        jsonrpc: '2.0',
+        method: 'queryCredentials',
+        params: {},
+      },
+    })) as Result<unknown>;
+
+    if (isError(res3)) {
+      throw new Error(res3.error);
+    }
+
+    expect(isSuccess(res3)).toBe(true);
+    expect(spyQuery2).toHaveBeenCalledTimes(1);
+
+    expect.assertions(8);
+  });
+
+  it('should fail changing queryPermission for another dApp', async () => {
+    const defaultState = getDefaultSnapState(account);
+
+    await snapMock.rpcMocks.snap_manageState({
+      operation: 'update',
+      newState: defaultState,
+    });
+
+    const res = (await onRpcRequest({
+      origin: 'https://random.io',
+      request: {
+        id: 'test-id',
+        jsonrpc: '2.0',
+        method: 'changePermission',
+        params: {
+          origin: 'https://random2.io',
+          method: 'queryCredentials',
+          value: false,
+        },
+      },
+    })) as Result<unknown>;
+
```

Comment:
```suggestion
    if (isError(ress)) {
      throw new Error(res.error);
    }
```

Reply:
Yes I know what it does, this `isError` would check if any error is thrown. It is good to have it there in case we update the tests and something breaks.